### PR TITLE
Fix Bugs

### LINKED ISSUE #155: [PE-D][Tester B] Product screenshot differs between UG and JAR
The UG does not currently reflect the current product. Perhaps it would be better to ensure that the UG reflects the latest version of the product release, rather than to state "to update".

## Steps to reproduce
`java -jar tahelper.jar`

## Product Screenshot when executing `JAR` file (Expected)

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: Javiery3889/ped#1

### LINKED ISSUE #183: [PE-D][Tester C] Ui not updating after search_student
command executed - /search_student name/J

feedback is correct indeed there are two students with J in their name but Ui is not updated and still listing all 4 students

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: dedsecrattle/ped#7

### LINKED ISSUE #192: [PE-D][Tester E] /search_student command not working after calling /list_students
Before: 

After executing command:

Steps to reproduce:
1. /add_student name/Adam email/adam@hotmail.com id/A1234567A
/add_student name/Blake email/blakey@gmail.com id/A0909091B
/add_student name/Carol email/cc@yahoo.com id/A7777777C

2. (Just to ascertain that the command works) call /search_student name/Adam. This should work as intended.

2. Call /list_students

3. Call /search_student name/Adam. The list does not change even though the logger states "1 person listed"

This is likely to be a GUI related bug where the view is not being updated.

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: marcus-ny/ped#4

### LINKED ISSUE #194: [PE-D][Tester D] Team names, courses and tutorial groups only visible upon restarting app
# Description
The user might not know the names of their randomised teams until the application is restarted. This significantly hinders the user as they receive a success message even though they are not able to immediately see the changes on the GUI.

## Steps to reproduce
Add students to randomised teams using the random teams command.
Try using the view tutorial command.

Restart the app and run the view tutorial command again.

## Expected
All teams should be visible before restarting the app and after restarting the app.

## Actual
The teams are only visible after restarting the app.

## Screenshots

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: itstrueitstrueitsrealitsreal/ped#20

### LINKED ISSUE #196: [PE-D][Tester A] Cannot tell the difference between students with the same name
/class_list_students module/CS2103T class/T09

There is a feature flaw for this command. Although students could be added successfully, there was no way to tell the difference between  students with the same name. This can make it difficult to know if the right person has been added since the unique identifier is their id or email, not their name. This can affect ability of app to be used for groups with same named members.

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: Darkarche3/ped#5

### LINKED ISSUE #203: [PE-D][Tester A] Search use is limited.
/search_student id/A0123456A - Feature Flaw. Signals number of people listed for id. However, as id is unique this is to be expected. Would make more sense if it gave a result which showed the person user is looking for especially since user might be searching a huge list

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: Darkarche3/ped#8

### LINKED ISSUE #205: [PE-D][Tester A] Cannot test view teams command.
/view_teams name/Team 1 module/CS2103T class/T09 as allocate team command not working. This is quite serious as it is one of the main features of the app. 

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: Darkarche3/ped#10

### LINKED ISSUE #207: [PE-D][Tester A] User Interface does not update immediately
When classes and modules are added, addition is successful but it is not reflected until the app is restarted or reopened. Makes it hard to use the app.

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: Darkarche3/ped#11

### LINKED ISSUE #208: [PE-D][Tester D] UG: Incorrect command format
# Description
The UG states that an example command is /class_list_students module/CS2103T class/T09,
whereas the help menu brings up that the correct format is /class_list_students module/CS2103T tutorial/T09
This is misleading to the user and may result in confusion.

## Steps to reproduce
Run the command `/class_list_students module/CS2103T class/T09` after having added students to the class T09 in the module CS2103T.

## Expected
Students are listed.

## Actual
The UI informs the user that the command format is incorrect.

## Screenshots

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: itstrueitstrueitsrealitsreal/ped#23

### LINKED ISSUE #214: [PE-D][Tester D] Team size not updated after deletion of student from team
# Description
After running the delete student from team command, using the view teams command results in an incorrect team member count, hindering the user from managing the teams accurately.

## Steps to reproduce
Delete a student from a team after having added them to the team, and use the view teams command to view the team.

## Expected
The team should have the updated size of the team logged.

## Actual
The team does not have the size updated.

## Screenshots

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: itstrueitstrueitsrealitsreal/ped#25

### LINKED ISSUE #219: [PE-D][Tester E] The class/module list GUI does not update after operations automatically (requires clicking the box/restarting)
Before:

After executing add command:

After restarting app:

Detailed description:

The module list does not update immediately after operations (example: tried with /add_class and /delete_class). Instead either a restart or clicking on the UI Box is required for the updates to be reflected on the GUI.

Steps to reproduce:

1. Launch the app
2. /add_class module/ACC1701X tutorial/T01

Output: Logger states "added ACC1701X T01". However, the GUI remains empty.

3. Either restart the app or click on the leftmost UI Box to get the updated GUI.

I believe this bug can be slightly problematic from a user standpoint as well. After adding a class, it would be more intuitive to see the new classes I've added visually without having to restart the app. Having to click on the GUI can also be unintuitive considering this is mainly meant to be a CLI app.

(Note: this issue may be interlinked with the student list not updating after /list_students command as well)

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: marcus-ny/ped#9

### LINKED ISSUE #220: [PE-D][Tester E] App may not feel very optimized for CLI usage
Detailed Description:

As stated in the User guide, the app is meant for CLI users and is optimized for that purpose. Admittedly, the usage of GUI features such as clicking to navigate through different modules and classes is fast and intuitive. However, as a CLI optimized app, I believe the app could support better CLI usage.

For instance, the need to click the GUI to navigate to each class's student list as shown here:

When the user wants to navigate back to the full student list, or to check a different class, the navigation is done through clicking the GUI rather than typing commands in the command line interface. 

Even though some commands have been provided to support these behaviours, the GUI does not seem to update automatically through CLI commands. It requires the user to click on the relevant tabs to refresh the list. This may defeat the purpose of making a CLI app as the users would need to use the GUI interactions more often and might slow down the process.

As the one of this project's purposes is to cater to fast typers to complete their desired tasks fast, I think the app could do with more commands to streamline this process. It might be violating project constraints which is that the app is supposed to be CLI-focused.

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: marcus-ny/ped#10

### LINKED ISSUE #221: [PE-D][Tester E] Lists on GUI (Module, Class and Student) do not sync up at certain points
Detailed Description:

The 3 existing lists - Module, Class and Student List do not sync up correctly sometimes.

(Note: this is not a functionality flaw/bug but more of a potential area of improvement for your feature)

Steps to reproduce:

1. Current list:
ACC1701X T03 has student Adamath in it.
ACC1701L T02 has no students.

2. Viewing ACC1701X T03 Class List.
a. Click ACC1701X in leftmost panel
b. Click T03 in middle panel.

3. Viewing ACC1701L
a. Click ACC1701L in leftmost panel

Here, ACC1701L has been selected and T02 is correctly displayed as the class belonging to that module. However, on the student list, Adamath is still there.

This may be misleading as it might seem as if Adamath is in ACC1701L T02 as well (while he isn't).

The need to click both ACC1701L then T02 means that a user would need to always click on the module and class to double check that the student list is updated correctly. Though not a very serious issue, this can feel slightly unintuitive and inconvenient.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: marcus-ny/ped#11

### LINKED ISSUE #222: [PE-D][Tester E] Empty class list is not shown correctly
Detailed description:

A class list of a class with zero students is not shown correctly.

Current state:
ACC1701L T02 is empty: (ascertained by /class_list_students module/ACC1701L tutorial/T02)

ACC1701X T03 has 2 students: (as shown by /class_list_students module/ACC1701X tutorial/T03)

Global Student List has 6 students:

Steps to reproduce:
1. Call /list_students (this will display the 6 students in the database)
2. Call /class_list_students module/ACC1701L tutorial/T02

Expected output: Since T02 of ACC1701L has no students, it should not show any students in rightmost panel.
Description: The global student list is displayed and the student list is not correctly shown as empty. To check if the GUI refresh issue from previous bugs is related, clicking on the rightmost panel box does not update the list. (Can be seen from the highlight on Adamath that the panel has been clicked and attempted to refresh)

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: marcus-ny/ped#12

### LINKED ISSUE #223: [PE-D][Tester E] View_teams command functionality is not very clear in UG
According to the UG, view_teams command seems to be intended for searching for a particular team in a tutorial and listing out the students in that team.

However, the UG documentation seems to state that both name and id are optional. However, running /view_teams module/<name> tutorial/<tutorial number> seems to return an error stating "Predicate type is missing".

Steps to reproduce:
1. Have one module and a corresponding tutorial in that mod. (ACC1701X T03)
2. /view_teams module/ACC1701X tutorial/T03

Moreover, the command word is "view_teams" which seems to indicate the ability to list out all the teams in a particular tutorial rather than the ability to find a team with a specific name or index. Perhaps "find_team" is more representative of this behaviour.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: marcus-ny/ped#13

### LINKED ISSUE #224: [PE-D][Tester E] View_teams command indexing starts from zero unlike all other commands which start from 1
Detailed Description:

When calling /view_teams index/<index> module/<> tutorial/<>, the index is zero based unlike other commands which were all 1-indexed.

Steps to reproduce:

Prerequisite: Have a module and a class (example: ACC1701X T03)
1. /add_team module/ACC1701X tutorial/T03 name/T1
2. /add_team module/ACC1701X tutorial/T03 name/T2
3. /view_teams index/2 module/ACC1701X tutorial/T03 (User wishes to see members in the second team in the list)

Expected output: User should be able to see details of team T2 since by index, T2 is the second team.

Actual output: Index out of range. 

More details: Using /view_teams index/1 module/ACC1701X tutorial/T03 returns the correct output and shows T02, indicating that zero indexing is used here.

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: marcus-ny/ped#14

### LINKED ISSUE #225: [PE-D][Tester D] Empty group not deleted
# Description
Deleting all students from a team and viewing the team results in a team of size 2 but no students being logged.

## Steps to reproduce
Delete all students from a team and run the view the team.

## Expected
Command failure as the group is empty

## Actual
The group is still logged as having their original size, but no students are listed.

## Screenshots

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: itstrueitstrueitsrealitsreal/ped#27

### LINKED ISSUE #226: [PE-D][Tester E] Select_Team GUI component does not update after forming teams
Steps to reproduce:

Precondition:
ACC1701X T03 has 6 students.

1. /random_teams module/ACC1701X tutorial/T03 teams/6.
Afterwards, the select_teams should look like this:

2. /random_teams module/ACC1701X tutorial/T03 teams/3

Expected output: Select_teams drop down should immediately change to 3 teams

Actual output: Team count remains at 6 from previous randomization. GUI component does not update automatically.

To refresh, user needs to click on ACC1701X course again to trigger the list to change to 3 teams.

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: marcus-ny/ped#15

### PR DESCRIPTION
**Pull request by qinxutan**
UI Fixes: 
Fixes #226 
Fixes #222
Fixes #221
Fixes #220
Fixes #219
Fixes #214 
Fixes #207 
Fixes #203
Fixes #194 
Fixes #192
Fixes #183 

Change Team Size to Team Capacity:
Fixes #225 

Fix indexing of View Team:
Fixes  #224 

UG Fixes:
Fixes #223
Fixes #208
Fixes #205
Fixes #155

Edit Command Output:
Fixes #196

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddTeamCommand.java [hunk 1]
```diff
@@ -120,7 +122,7 @@ private String generateSuccessMessage(ModuleCode module, TutorialClass tutorialS
      * @return The success message.
      */
     private String generateSuccessMessage(ModuleCode module, TutorialClass tutorialString, String teamName,
-            int teamSize) {
+                                          int teamSize) {
```
Comment (whelan-low) [inline src/main/java/seedu/address/logic/commands/AddTeamCommand.java:125]: Is the whitespace here necessary?

**Review by whelan-low — CHANGES_REQUESTED** [review]
LGTM so far! However, just want to check for the rest of the commands that don't seem to be changed:
Add, Edit, Delete student (from TAHelper itself): Do they work as intended with the UI?
Delete module: Does the module pane need to be updated? (Same for Add/Delete class)

**Comment by qinxutan** [comment]
> LGTM so far! However, just want to check for the rest of the commands that don't seem to be changed: Add, Edit, Delete student (from TAHelper itself): Do the student details need to be updated in the UI? Delete module: Does the module pane need to be updated? (Same for Add/Delete class)

yep the module pane updates for delete module/add/delete class under the usemoduleview
as for the student, the personlist is updated when thrs any changes made to student and then usepersonview

File: src/main/java/seedu/address/logic/commands/AddTeamCommand.java (hunk 1 above)
Comment (qinxutan) [reply src/main/java/seedu/address/logic/commands/AddTeamCommand.java:125]: ok will fix!

**Pull request merged**